stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageComposerViewModel.kt

```diff
@@ -149,46 +87,24 @@ public class MessageComposerViewModel(
      *
      * @param attachment The attachment to remove.
      */
-    public fun removeSelectedAttachment(attachment: Attachment) {
-        this.selectedAttachments = this.selectedAttachments - attachment
-    }
-
-    /**
-     * Clears all the data from the input - both the current [input] value and the
-     * [selectedAttachments].
-     */
-    private fun clearData() {
-        input = ""
-        selectedAttachments = emptyList()
-    }
+    public fun removeSelectedAttachment(attachment: Attachment): Unit =
+        messageComposerController.removeSelectedAttachment(attachment)
 
     /**
-     * Sends a given message using our Stream API. Based on [isInEditMode], we either edit an existing
-     * message, or we send a new message, using the [ChatDomain].
+     * Sends a given message using our Stream API. Based on the [MessageComposerController] state, we either edit an
+     * existing message, or we send a new message, using the [ChatDomain].
      *
      * It also dismisses any current message actions.
      *
      * @param message The message to send.
      */
-    public fun sendMessage(message: Message) {
-        viewModelScope.launch {
-            val sendMessageCall = if (isInEditMode) {
-                chatDomain.editMessage(message)
-            } else {
-                chatDomain.sendMessage(message)
-            }
-
-            dismissMessageActions()
-            sendMessageCall.enqueue()
-        }
-        clearData()
-    }
+    public fun sendMessage(message: Message): Unit = messageComposerController.sendMessage(message)
 
     /**
-     * Builds a new [Message] to send to our API. If [isInEditMode] is true, we use the current
+     * Builds a new [Message] to send to our API. Based on the [MessageComposerController] state, we use the current
```

Comment:
`MessageComposerController` should probably not show up in any of the docs of this class. It's an implementation detail, users should not know that it exists as far as I understand.